src/port1.0/portlint.tcl

```diff
@@ -562,15 +590,12 @@ proc portlint::lint_main {args} {
         incr warnings
     }
 
-    if {[info exists platforms]} {
-        foreach platform $platforms {
-            set platname [lindex $platform 0]
-            if {$platname ni $lint_platforms} {
-                ui_error "Unknown platform: $platname"
-                incr errors
-            } else {
-                ui_info "OK: Found platform: $platname"
-            }
+    if {$seen_platforms && [info exists platforms]} {
+        set platforms_lint [portlint::lint_platforms $platforms]
+
+        foreach err $platforms_lint {
+            ui_error $err
+            incr errors
```

Comment:
Redundantly setting `platforms` to the default value should not be an error, just a warning.